packages/ui/src/components/filters/manage-views.tsx

```diff
@@ -0,0 +1,225 @@
+      const exists = trimmedName && existingNames.has(trimmedName) && trimmedName !== view.name.trim().toLowerCase()
+      const hasValidationError = exists || !trimmedName
+
+      setHasError(hasValidationError)
+      // Always update parent to reflect current state
+      onRename(view.id, newName)
+    },
+    [view.id, view.name, existingNames, onRename]
+  )
+
+  return (
+    <div
+      ref={setNodeRef}
+      style={{ transform: CSS.Transform.toString(transform), transition }}
+      className="flex items-center gap-x-2 p-2 hover:bg-background-4 rounded"
+    >
+      <div className="cursor-move group" {...attributes} {...listeners}>
+        <Icon
+          className="text-icons-4 group-hover:text-icons-2 transition-colors duration-200"
+          name="grid-dots"
+          size={14}
+        />
+      </div>
+
+      <Input
+        theme={hasError ? 'danger' : 'default'}
+        value={name}
+        size="md"
+        placeholder="Enter a name"
+        onChange={handleChange}
+      />
+
+      <button
+        className="text-icons-4 hover:text-icons-danger"
+        onClick={() => onDelete(view.id)}
+        aria-label="Delete view"
+      >
+        <Icon className="rotate-45" name="plus" size={14} />
+      </button>
+    </div>
+  )
+})
+
+SortableItem.displayName = 'SortableItem'
+
+interface ManageViewsProps {
+  open: boolean
+  onOpenChange: (open: boolean) => void
+  views: SavedView[]
+  onSave: (views: SavedView[]) => void
+  onRename: (id: string, newName: string) => void
+}
+
+const ManageViews: FC<ManageViewsProps> = memo(({ open, onOpenChange, views, onSave }) => {
+  const [localViews, setLocalViews] = useState<SavedView[]>([])
+
+  /**
+   * Updates local state when dialog opens to:
+   * - Create an isolated copy of views for local editing
+   * - Allow cancellation of changes
+   * - Prevent parent state updates until save
+   */
+  useEffect(() => {
+    if (open) setLocalViews(views)
+  }, [open, views])
+
+  const existingNames = useMemo(() => new Set(localViews.map(v => v.name.trim().toLowerCase())), [localViews])
+
+  /**
+   * Handlers for view modifications
+   * Memoized to prevent unnecessary re-renders of child components
+   */
+  const handleLocalRename = useCallback((viewId: string, newName: string) => {
+    setLocalViews(prev => prev.map(view => (view.id === viewId ? { ...view, name: newName } : view)))
+  }, [])
+
+  const handleLocalDelete = useCallback((viewId: string) => {
+    setLocalViews(prev => prev.filter(v => v.id !== viewId))
+  }, [])
+
+  const { handleDragEnd, getItemId } = useDragAndDrop({
+    items: localViews,
+    onReorder: setLocalViews
+  })
+
+  /**
+   * Validates the current state of views
+   * Checks for empty names and duplicates
+   */
+  const hasErrors = useCallback(() => {
+    const duplicatesMap = new Map<string, boolean>()
+
+    return localViews.some(view => {
+      const trimmedName = view.name.trim().toLowerCase()
+
+      // Проверка на пустое имя
```

Comment:
i'd like to avoid non-english comments in the codebase. open to opinions though.